scenario-compiler/src/main/scala/pl/touk/nussknacker/engine/extension/CastOrToConversionExt.scala

```diff
@@ -0,0 +1,165 @@
+package pl.touk.nussknacker.engine.extension
+
+import cats.data.ValidatedNel
+import cats.implicits.catsSyntaxValidatedId
+import pl.touk.nussknacker.engine.api.generics.{GenericFunctionTypingError, MethodTypeInfo, Parameter}
+import pl.touk.nussknacker.engine.api.typed.typing
+import pl.touk.nussknacker.engine.api.typed.typing.{Typed, TypedObjectWithValue, TypingResult, Unknown}
+import pl.touk.nussknacker.engine.definition.clazz.{ClassDefinitionSet, FunctionalMethodDefinition, MethodDefinition}
+import pl.touk.nussknacker.engine.extension.CastOrToConversionExt.getConversion
+import pl.touk.nussknacker.engine.util.Implicits.RichScalaMap
+import pl.touk.nussknacker.engine.util.classes.Extensions.{ClassExtensions, ClassesExtensions}
+
+import java.lang.{Boolean => JBoolean}
+import scala.util.Try
+
+// todo: lbg - add casting methods to UTIL
+class CastOrToConversionExt(target: Any, classesBySimpleName: Map[String, Class[_]]) {
+
+  def is(className: String): Boolean =
+    getClass(className).exists(clazz => clazz.isAssignableFrom(target.getClass)) ||
+      getConversion(className).exists(_.canConvert(target))
+
+  def to(className: String): Any = tryCast(className) match {
+    case Right(value) => value
+    case Left(ex1) =>
```

Comment:
Let's replace this with `... orElse ... getOrElse ...` chain

Reply:
Done slightly different